### PR TITLE
[row-count][l]: calculates rows separately from data fetching.

### PR DESCRIPTION
**Pull request by EvgeniiaVak**
* merges in EDS `datapackage-views-js` and EDS `datastore-query-builder` (can move back out, but why? these branches are used for frontend-eds only and it only uses data-explorer together)
* executes SQL for counting rows in a separate call from preview data fetching =>
  * all fist dataset page loading times are faster
  * if row count query still fails because of the timeout (it can in theory), the page and preview still loads (although without row count)
* removes page numbers from pagination (because we do not always know the last page now), leaves out buttons for `first`, `next`, `previous` and sometimes (when the row count loads) `last` pages.

### CODE REVIEW

File: src/utils/loadDataset.js [hunk 1]
```diff
@@ -19,6 +19,8 @@ export default async dpID => {
   try {
     const dataset = await Dataset.load(DP_ID)
 
+    // console.log('dataset: ' + JSON.stringify(dataset, null, 2))
+
```
Comment (leomrocha) [inline src/utils/loadDataset.js:23]: do we need the comment or can be cleaned up?

File: src/utils/countRows.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    } else {
+
+      // this query is for optimizing the row-count calculation for non-filtered data only
+      // more on how it works here https://www.citusdata.com/blog/2016/10/12/count-performance/#distinct_counts_exact_index
+      rowCountSql = `SELECT COUNT(*) AS _count FROM (SELECT DISTINCT "_id" FROM "${name}") t`
+
+    }
+
+    // console.log(`rowCountSql: ` + rowCountSql)
```
Comment (leomrocha) [inline src/utils/countRows.js:33]: can we cleanup the comments?

File: src/components/QueryBuilder/QueryBuilder.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      let pathParts = urlObj.pathname.split('/')
+      pathParts.pop()
+      urlObj.pathname = pathParts.join('/') + '/' // Trailing slash for consistency
+      urlObj.search = '' // Remove all search params
+      apiUrl = urlObj.href
+    } else {
+      apiUrl = (new URL(resource.path)).origin + '/api/3/action/'
+    }
+
+    // TODO: remove rowcout?
```
Comment (leomrocha) [inline src/components/QueryBuilder/QueryBuilder.js:22]: Does it need to be done from the return object?

**Review by leomrocha — APPROVED** [review]
Please check the comments

File: src/utils/loadDataset.js (hunk 1 above)
Comment (EvgeniiaVak) [reply src/utils/loadDataset.js]: :+1: cleaned up

File: src/components/QueryBuilder/QueryBuilder.js (hunk 3 above)
Comment (EvgeniiaVak) [reply src/components/QueryBuilder/QueryBuilder.js]: :+1: cleaned up

**Pull request merged**